openfl/interface/interactive_api/experiment.py

```diff
@@ -423,6 +427,41 @@ def decorator_with_args(training_method):
 
         return decorator_with_args
 
+    def set_aggregation_function(self, aggregation_function: AggregationFunctionInterface):
+        """Set aggregation function for the task.
+
+        Args:
+            aggregation_function: Aggregation function.
+
+        You might need to override default FedAvg aggregation with built-in aggregation types:
+            - openfl.component.aggregation_functions.GeometricMedian
+            - openfl.component.aggregation_functions.Median
+        or define your own AggregationFunctionInterface subclass.
+        See more details on `Overriding the aggregation function`_ documentation page.
+        .. _Overriding the aggregation function:
+            https://openfl.readthedocs.io/en/latest/overriding_agg_fn.html
+        """
+        def decorator_with_args(training_method):
+            if not isinstance(aggregation_function, AggregationFunctionInterface):
+                raise Exception('aggregation_function must implement '
+                                'AggregationFunctionInterface interface.')
+            self.aggregation_functions[training_method.__name__] = aggregation_function
+            return training_method
```

Comment:
Try to use `@functools.wraps`
https://docs.python.org/3/library/functools.html#functools.wraps

Reply:
Added `functools.wraps` just like in `register_fl_task` decorator.

Reply:
It looks like in `register_fl_task` it doesn't work and should be fixed.

Reply:
Removed `functools` decorators for `TaskInterface`.